### ISSUE TITLE
[Progress bar] Option with no visible label should have `aria-label`

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3626]
### 😯 Current Behavior

Progress bars with no visible label currently have `aria-label` added to `<sl-progress-bar>`. This is not correct, because `aria-label` should be added to `<div role='progressbar'>`. Only then `aria-label` will be properly announced by screen readers.

<img width="973" height="413" alt="Image" src="https://github.com/user-attachments/assets/d5e56be9-dd1d-47af-9b43-f0a5982115a4" />

### 🤔 Expected Behavior/possible solution?

`aria-label` should be removed from `<sl-progress-bar>` and added to `<div role='progressbar'>`.

Code for progress bar with no visible label should look like:
```
<sl-progress-bar>
    <div aria-label="Progress bar label in the indeterminate state" aria-describedby="helper" class="container" role="progressbar" aria-valuemin="0" aria-valuemax="100"></div>
<sl-progress-bar>
```

### 🖥️ Steps to Reproduce

1. Go to https://storybook.sanomalearning.design/?path=/story/feedback-status-progress-bar--all
2. Verify if storybook's a11y plugin reports errors called "ARIA progressbar name"

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/feedback-status-progress-bar--all

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

All

### Component version

0.2.1

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning
